modules/apps/29-fee/keeper/escrow_test.go

```diff
@@ -122,66 +122,94 @@ func (suite *KeeperTestSuite) TestEscrowPacketFee() {
+				suite.Require().Equal(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(0)), balance)
 			},
 		},
 		{
-			"invalid forward address", func() {
+			"invalid forward address",
+			func() {
 				forwardRelayer = "invalid address"
 			},
 			func() {
-				// check if the refund acc has been refunded the timeoutFee & onRecvFee
-				expectedRefundAccBal := refundAccBal.Add(fee.TimeoutFee[0]).Add(fee.RecvFee[0]).Add(fee.TimeoutFee[0]).Add(fee.RecvFee[0])
-				hasBalance := suite.chainA.GetSimApp().BankKeeper.HasBalance(suite.chainA.GetContext(), refundAcc, expectedRefundAccBal)
-				suite.Require().True(hasBalance)
-
+				// check if the refund acc has been refunded the timeoutFee & recvFee
+				expectedRefundAccBal := refundAccBal.Add(defaultTimeoutFee[0]).Add(defaultReceiveFee[0]).Add(defaultTimeoutFee[0]).Add(defaultReceiveFee[0])
+				balance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(expectedRefundAccBal, balance)
 			},
 		},
 		{
-			"invalid forward address: blocked address", func() {
+			"invalid forward address: blocked address",
+			func() {
 				forwardRelayer = suite.chainA.GetSimApp().AccountKeeper.GetModuleAccount(suite.chainA.GetContext(), transfertypes.ModuleName).GetAddress().String()
 			},
 			func() {
-				// check if the refund acc has been refunded the timeoutFee & onRecvFee
-				expectedRefundAccBal := refundAccBal.Add(fee.TimeoutFee[0]).Add(fee.RecvFee[0]).Add(fee.TimeoutFee[0]).Add(fee.RecvFee[0])
-				hasBalance := suite.chainA.GetSimApp().BankKeeper.HasBalance(suite.chainA.GetContext(), refundAcc, expectedRefundAccBal)
-				suite.Require().True(hasBalance)
+				// check if the refund acc has been refunded the timeoutFee & recvFee
+				expectedRefundAccBal := refundAccBal.Add(defaultTimeoutFee[0]).Add(defaultReceiveFee[0]).Add(defaultTimeoutFee[0]).Add(defaultReceiveFee[0])
+				balance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(expectedRefundAccBal, balance)
+			},
+		},
+		{
+			"invalid receiver address: ack fee returned to sender",
+			func() {
+				reverseRelayer = suite.chainA.GetSimApp().AccountKeeper.GetModuleAccount(suite.chainA.GetContext(), transfertypes.ModuleName).GetAddress()
+			},
+			func() {
+				// check if the refund acc has been refunded the timeoutFee & ackFee
+				expectedRefundAccBal := refundAccBal.Add(defaultTimeoutFee[0]).Add(defaultAckFee[0]).Add(defaultTimeoutFee[0]).Add(defaultAckFee[0])
+				balance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAcc, sdk.DefaultBondDenom)
+				suite.Require().Equal(expectedRefundAccBal, balance)
 			},
 		},
+		// {
+		// 	"invalid refund address: no-op, timeout fee remains in escrow",
```

Comment:
I'll see if I can fix this test. Currently failing because we're relying on `EscrowPacketFee` and `malleate()` is called after this. If we call `malleate()` before escrowing the fees, then the `RefundAddress` (incentivizer) is the module account, which doesn't have balance.

Bit of a mouth full, but hope it makes sense.

Reply:
The simplest solution is to mutate `packetFee.RefundAddress` in `malleate()` after the fees are initially escrowed and before `DistributePacketFees` is called 👍